mmedit/models/backbones/sr_backbones/cain_net.py

```diff
@@ -0,0 +1,386 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from mmcv.runner import load_checkpoint
+
+from mmedit.models.common import make_layer
+from mmedit.models.registry import BACKBONES
+from mmedit.utils import get_root_logger
+
+
+def pixel_shuffle(x, scale=8, up=True):
+    """Up-scale or down-scale by pixel shuffle.
+
+    Args:
+        x (Tensor): Input tensor.
+        scale (int): Scale factor.
+        up (bool): Up-scale or down-scale.
+
+    Returns:
+        y (Tensor): Output tensor.
+    """
+
+    b, c, h, w = x.shape
+    if up:
+        y = F.pixel_shuffle(x, scale)
+    else:
+        h = int(h / scale)
+        w = int(w / scale)
+        x = x.view(b, c, h, scale, w, scale)
+        shuffle_out = x.permute(0, 1, 3, 5, 2, 4).contiguous()
+        y = shuffle_out.view(b, c * scale * scale, h, w)
```

Comment:
How about we put this as a separate function, and store it in `mmediting/mmedit/models/common/`? That could be useful for other methods. 

Also, we usually call this "pixel_unshuffle". So it's better to put it in a separate function.